### DISCUSSION TITLE
"`opts.silent = true` by default" not working?

### DISCUSSION THREAD

**Discussion by xfzv** [mrjones2014/legendary.nvim#350] [Q&A] [answered]
https://github.com/mrjones2014/legendary.nvim/blob/b8cb553dff63a1ded1be1f258cf8e0e4c443e302/README.md?plain=1#L110

Not sure why but in my case, if I don't explicitly use `opts = { silent = true }` for a keymap, it's echoed to command line.

<details>
  <summary>Minimal init.lua to reproduce</summary>

```lua
vim.cmd([[set runtimepath=$VIMRUNTIME]])
vim.cmd([[set packpath=/tmp/nvim/site]])
local package_root = "/tmp/nvim/site/pack"
local install_path = package_root .. "/packer/start/packer.nvim"
vim.g.loaded_remote_plugins = ""
local function load_plugins()
	require("packer").startup({
		{
			"wbthomason/packer.nvim",
			{
				"mrjones2014/legendary.nvim",
				requires = {
					{
						"nvim-telescope/telescope.nvim",
						requires = "nvim-lua/plenary.nvim",
					},
					"stevearc/dressing.nvim",
				},
			},
			{
				"nvim-neo-tree/neo-tree.nvim",
				requires = {
					"MunifTanjim/nui.nvim",
				},
			},
		},
		config = {
			package_root = package_root,
			compile_path = install_path .. "/plugin/packer_compiled.lua",
			display = { non_interactive = true },
		},
	})
end
_G.load_config = function()
	require("telescope").setup()
	require("dressing").setup()
	require("legendary").setup({
		keymaps = {
			{
				"<leader>n",
				":Neotree show toggle <CR>",
				-- opts = { silent = true },
				description = "NeoTree",
			},
		},
	})
	require("neo-tree").setup({})
end
if vim.fn.isdirectory(install_path) == 0 then
	print("Installing legendary.nvim and dependencies.")
	vim.fn.system({ "git", "clone", "--depth=1", "https://github.com/wbthomason/packer.nvim", install_path })
end
load_plugins()
require("packer").sync()
vim.cmd([[autocmd User PackerComplete ++once echo "Ready!" | lua load_config()]])
  ```
  
</details>

1. Run `nvim --clean -u init.lua` (as suggested [here](https://github.com/mrjones2014/legendary.nvim/discussions/93#discussioncomment-4109260))
2. Run `:Legendary`
3. Select the NeoTree mapping and press "CR" to run it
![image](https://user-images.githubusercontent.com/78810647/230714844-35366e48-8a1c-467d-adaf-72faa3e5f79c.png)
4. The command is echoed to command line:
![image](https://user-images.githubusercontent.com/78810647/230714595-880ea917-bb66-4d7f-a513-20381d7df4e2.png)

5. Repeat 1-3 but uncomment `opts = { silent = true }`:
![image](https://user-images.githubusercontent.com/78810647/230714597-15f3cb2f-f4a9-4863-af39-64a6ce090d18.png)

Am I missing something?

Thanks!

**Comment by mrjones2014** [comment, marked as answer]
I don’t think you’re doing anything wrong, I think that bit of my docs is inaccurate, sorry. You can set `default_opts` for each item type via config.

**Reply by mrjones2014** [reply]
Updated the docs and fixed.
